locator/token_metadata.hh

```diff
@@ -72,6 +72,50 @@ struct host_id_or_endpoint {
 class token_metadata_impl;
 struct topology_change_info;
 
+struct version_tracker {
+public:
+    friend class shared_token_metadata;
+
+    using link_base = boost::intrusive::list_member_hook<boost::intrusive::link_mode<boost::intrusive::auto_unlink>>;
+    struct link_type : link_base {
+        link_type() noexcept = default;
+        link_type(link_type&& o) noexcept { swap_nodes(o); }
+        link_type& operator=(link_type&& o) noexcept {
+            if (this != &o) {
+                this->~link_type();
+                new(this) link_type(std::move(o));
+            }
```

Comment:
Please avoid this style of operator=, I have doubts about whether it is legal. The destructor ends the lifetime of the object, but older references to it don't now that.

Reply:
Fixed.